mlir/lib/Dialect/Affine/Analysis/Utils.cpp

```diff
@@ -50,6 +52,9 @@ void LoopNestStateCollector::collect(Operation *opToWalk) {
       loadOpInsts.push_back(op);
     else if (isa<AffineWriteOpInterface>(op))
       storeOpInsts.push_back(op);
+    else if (!mlir::isMemoryEffectFree(op) &&
+             !isa<mlir::memref::AllocOp, mlir::memref::DeallocOp>(op))
```

Comment:
Please don't hardcode alloc, dealloc; instead check for `Alloc` and `Free` effects.